documentation/graphing-overview/index.html.md

```diff
@@ -0,0 +1,279 @@
+
+	$("#submitButton").click(function(){
+		Flow.channel.operations.publish('updateXY', [$("#x").val(), $("#y").val()]);
+	});
+
+We use the `operations.publish()` to guarantee than any subscribers are notified of the call. (Using [`Flow.channel.run.do()`](../../api_adapters/generated/run-api-service/) also calls the operation, but does not update subscribers.)
+	
+Finally, use the Flow.js [Operations Channel](../generated/channels/operations-channel/) to subscribe to the operation you've just called. The second argument is a callback function, called whenever the model operation is called. We want to update and re-render our visualization each time this operation happens:
+
+	Flow.channel.operations.subscribe('updateXY',
+		function(data) {
+			var composedData = {
+				x: data.updateXY.arguments[0],
+				y: data.updateXY.arguments[1]
+			};
+			myThirdChartData.push(composedData);
+			myThirdChart.setData(myThirdChartData);
+			myThirdChart.render();
+	});
+
+The argument passed to the callback (`data`) is an object with the names and values of the arguments used to call the operation. The variable `composedData` uses these values to create an (x,y) data point to add to the data set we're graphing.
+
+
+<a name="example"></a>
+**Putting it all together.**
+
+Here's the complete sample code:
+
+	<html>
+
+	<head>
+
+		<!-- for Contour -->
+		<link rel="stylesheet" href="http://forio.com/tools/contour/contour.min.css">
+		<script src="http://d3js.org/d3.v3.js"></script>
+		<script src="http://cdnjs.cloudflare.com/ajax/libs/lodash.js/2.4.1/lodash.js"></script>
+		<script src="https://forio.com/tools/contour/contour.js"></script>
+
+		<!-- for Flow.js -->
+		<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.2/jquery.min.js"></script>
+		<script src="https://forio.com/tools/js-libs/1.5.0/epicenter.min.js"></script>
+		<script src="https://forio.com/tools/js-libs/flow/0.9.0/flow.js"></script>
+
+	</head>
+
+	<body>
+
+		<p>Here is a div with the Contour chart, used in Steps 1-4: </p>
+		<div class="myChart"></div>
+		<p>The next data point is added to the chart each time the user enters a new value:</p>
+		<input data-f-bind="sampleInt"></input>
+		<hr>
+
+		<p>Here is a div with the Contour chart used in Extension #1, based on two variables updated automatically when the model is advanced: </p>
+		<div class="mySecondChart"></div>
+		<p><button data-f-on-click='advanceModel'>Advance Model</button></p>
+		<hr>
+		
+		<p>Here is a div with the Contour chart used in Extension #2, based on two variables whose values are entered by the user: </p>
+		<div class="myThirdChart"></div>
+		<p>
+			Enter x-coordinate: <input id="x"></input><br>
+			Enter y-coordinate: <input id="y"></input></br>
+			<button id="submitButton">Update x,y</button>
+		</p>
+		
+		<script>
+
+			Flow.initialize({
+				channel: {
+					run: { 
+						model: 'supply-chain-game.py',
+						account: 'acme-simulations',
```

Comment:
Add a comment here that you only need the account and project if you're running locally, Michael purvis didn't realize that and had to change his code each time he deployed to a different url